### PR TITLE
KTX Compare Tool - Phase 2

### PR DESCRIPTION
**Pull request by aqnuep**
This now includes the complete implementation for the compare tool, as of the baseline, but once this is merged to the ktx-compare branch and that branch will result in a rebased PR against main, then I expect some minor changes.

This change also solves the issues that resulted in invalid golden files, because some of the compare tool testing depended on fixing the cause of those.

### CODE REVIEW

**Comment by aqnuep** [comment]
@MarkCallow, please review this change indifferent of the fact that it is a draft and CI may not pass yet.

This will anyway first go into the `ktx-compare` branch for finalization, rebasing, etc. before the final PR to main.

Thanks!

File: tools/ktx/command_create.cpp
Comment (MarkCallow) [inline tools/ktx/command_create.cpp:550]: Can we not duplicate these long strings?

**Review by MarkCallow — CHANGES_REQUESTED** [review]
Well-spotted on the enum bitfields. Which compiler(s) had problems with them? I thought enums were usually ints under the covers so I'm surprised those definitions were not cross platform.

I spotted a couple of small things I'd like  you to change.

File: tools/imageio/imageio_utility.h
Comment (aqnuep) [reply tools/imageio/imageio_utility.h:132]: memcpy guarantees bitwise copy, whereas other methods may hit undefined behavior land, hence generally these case are handled as such in most places. For small data this should be optimized out anyway.

File: tools/imageio/formatdesc.h
Comment (aqnuep) [reply tools/imageio/formatdesc.h]: Refactored.

File: tools/ktx/command_create.cpp
Comment (aqnuep) [reply tools/ktx/command_create.cpp]: Refactored.

File: tools/imageio/imagespan.hpp
Comment (aqnuep) [reply tools/imageio/imagespan.hpp]: Fixed.

**Comment by aqnuep** [comment]
Hi @MarkCallow! Do you need more time to review this or the related CTS change or should I proceed with merging these into the ktx-compare development branch?

**Comment by MarkCallow** [comment]
> Do you need more time to review this or the related CTS change

I just attempted a review. Please see my comment over there.

**Comment by MarkCallow** [comment]
> > Do you need more time to review this or the related CTS change
> 
> I just attempted a review. Please see my comment over there.

The changes in this PR are fine. I am in the midst of a very busy period. I won't have time to start trying `ktx compare` until the middle of next week. I don't want to hold you up any more so go ahead and merge the CTS changes and this.

**Comment by aqnuep** [comment]
Thank you, @MarkCallow. I'll get everything merged in the ktx-compare development branch and then we'll start rebasing/integrating mainline chaines into that development branch, because that'll take some time too. After that I still intend to open PRs before merging the development branch into main, so you'll have the chance to do another review round and testing, as necessary.

**Comment by MarkCallow** [comment]
> I'm getting this error message:
> 
> ...

This is a recurring issue with Chocolatey. It is one of the reasons CI tries 4 times to download Doxygen before failing. I do not fully understand it and have not had much help from GitHub. I just tried re-running but it failed again. I suggest trying to re-run tomorrow.

**Pull request merged**
